app/lib/features/chat/widgets/custom_input.dart

```diff
@@ -692,10 +705,23 @@ class _TextInputWidgetConsumerState extends ConsumerState<_TextInputWidget> {
   @override
   void initState() {
     super.initState();
-    WidgetsBinding.instance.addPostFrameCallback((_) {
-      widget.controller.text = ref.read(
-        chatInputProvider.select((value) => value.message),
-      );
+    ref.listenManual(
+        chatInputProvider.select((state) => state.selectedMessageState),
+        (prev, next) {
+      if (next == SelectedMessageState.edit) {
+        // a new message has been selected to be edited or switched from reply
+        // to edit, force refresh the inner text controller to reflect that
+        widget.controller.text = parseEditMsg(ref);
+        // frame delay to keep focus connected with keyboard.
+        WidgetsBinding.instance.addPostFrameCallback((_) {
+          widget.chatFocus.requestFocus();
+        });
+      } else if (next == SelectedMessageState.replyTo) {
+        // frame delay to keep focus connected with keyboard..
+        WidgetsBinding.instance.addPostFrameCallback((_) {
+          widget.chatFocus.requestFocus();
+        });
+      }
```

Comment:
isn't it easier to just wrap the entire part in the PostFrameCallback? It's also more legible...
```suggestion
        WidgetsBinding.instance.addPostFrameCallback((_) {
      if (next == SelectedMessageState.edit) {
        // a new message has been selected to be edited or switched from reply
        // to edit, force refresh the inner text controller to reflect that
        widget.controller.text = parseEditMsg(ref);
        // frame delay to keep focus connected with keyboard.
          widget.chatFocus.requestFocus();
      } else if (next == SelectedMessageState.replyTo) {
        // frame delay to keep focus connected with keyboard..
          widget.chatFocus.requestFocus();
      }
        });
```

Reply:
It can be. Thanks for pointer!

Reply:
Unfortunately, it throws error because ```ref``` update call. We have to keep it as before as ref isn't valid in next frame...
```
════════ Exception caught by scheduler library ═════════════════════════════════
Bad state: Cannot use "ref" after the widget was disposed.

```